src/construction.h

```diff
@@ -28,6 +28,7 @@ struct construction {
         std::string pre_note; // Additional note displayed along with construction requirements.
         std::string pre_terrain; // beginning terrain for construction
         std::string post_terrain;// final terrain after construction
+        std::string byproduct_item_group; // item group of byproducts created by the construction
```

Comment:
Would be nice if you could wrap this into `cata::optional` instead of using a sentinel value (empty string). That would show it's optional and we could move away from using arbitrary sentinel values.

Reply:
Done. I think I've done it as expected--it's my first encounter with `cata::optional`.